src/core/src/FabricMaterial.cpp

```diff
@@ -1022,6 +1070,8 @@ void FabricMaterial::reset() {
 
     _properties.clear();
 
+    destroyConnectionsToProperties();
```

Comment:
It's cleaner to destroy connections in `reset` and create connections in `setMaterial`

Reply:
Cleaner, probably yes, but what is the performance impact?

Reply:
It should be the same. `reset` and `setMaterial` get called the same number of times so this is just moving the call to a different spot. And it's zero cost if the tile doesn't have metadata.